packages/core/src/store/reducer.ts

```diff
@@ -71,9 +117,14 @@ export interface State {
   plugins: Record<PluginType, Plugin<any>>
   documents: Record<string, PluginState>
   focus?: string
+  history?: {
+    initialState: State
+    actions: Undoable[][]
```

Comment:
Where do you handle the grouping of actions?

Reply:
I don't yet, but would be handled in the `save` function

Reply:
Makes sense, wasn't sure if this was already part of the PR or not. So we basically commit every action right now (all action arrays of length 1) but we could add a more sophisticated behavior later. Sounds good